solr/core/src/java/org/apache/solr/handler/admin/api/CreateCore.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.handler.admin.api;
+
+import static org.apache.solr.common.params.CollectionAdminParams.PROPERTY_PREFIX;
+import static org.apache.solr.handler.api.V2ApiUtils.flattenMapWithPrefix;
+import static org.apache.solr.handler.api.V2ApiUtils.flattenToCommaDelimitedString;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import jakarta.inject.Inject;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.solr.client.api.endpoint.CreateCoreApi;
+import org.apache.solr.client.api.model.CreateCoreRequestBody;
+import org.apache.solr.client.api.model.CreateCoreResponseBody;
+import org.apache.solr.client.api.model.SolrJerseyResponse;
+import org.apache.solr.cloud.ZkController;
+import org.apache.solr.common.params.CoreAdminParams;
+import org.apache.solr.common.util.PropertiesUtil;
+import org.apache.solr.common.util.StrUtils;
+import org.apache.solr.core.CoreContainer;
+import org.apache.solr.handler.admin.CoreAdminHandler;
+import org.apache.solr.jersey.PermissionName;
+import org.apache.solr.jersey.SolrJacksonMapper;
+import org.apache.solr.request.SolrQueryRequest;
+import org.apache.solr.response.SolrQueryResponse;
+import org.apache.solr.security.PermissionNameProvider;
+
+/**
+ * Implementation of V2 api {@link CreateCoreApi} for creating core.
+ *
+ * @see CreateCoreRequestBody
+ * @see CoreAdminAPIBase
+ */
+public class CreateCore extends CoreAdminAPIBase implements CreateCoreApi {
+  private final ObjectMapper objectMapper;
+
+  @Inject
+  public CreateCore(
+      CoreContainer coreContainer,
+      CoreAdminHandler.CoreAdminAsyncTracker coreAdminAsyncTracker,
+      SolrQueryRequest req,
+      SolrQueryResponse rsp) {
+    super(coreContainer, coreAdminAsyncTracker, req, rsp);
+    this.objectMapper = SolrJacksonMapper.getObjectMapper();
+  }
+
+  @Override
+  @PermissionName(PermissionNameProvider.Name.CORE_EDIT_PERM)
+  public SolrJerseyResponse createCore(
+      CreateCoreRequestBody createCoreRequestBody, String coreName) {
+    final Map<String, Object> requestBodyMap =
```

Comment:
[0] I wrote and have used `Utils.reflectToMap()` to do similar things for other Solr APIs.  I'm not an expert on either Jackson or reflection - so I can't really tell whether this is the same or different, better or worse than how `reflectToMap` works.  Do you happen to know?  If using Jackson directly in this way is better, maybe it'd make sense to reimplement `reflectToMap` to do the better thing and we could call it here.

Alternately if you just weren't aware of `Utils.reflectToMap`, maybe the quickest path forward is to just use reflectToWrite here, and then we can tackle evaluating and potentially reimplementing `Utils.reflectToMap` as a separate JIRA ticket later.